DESIGN.md

```diff
@@ -325,18 +325,158 @@ In the addition of the previous rules, an operator for `Flowable`:
 
 ### Creation
 
-Creation of the various types should be exposed through factory methods that provide safe construction.
+Unlike RxJava 1.x, 2.x base classes are to be abstract, stateless and generally no longer wrap an `OnSubscribe` callback - this saves allocation in assembly time without limiting the expressiveness. Operator methods and standard factories still live as final on the base classes.
+
+Instead of the indirection of an `OnSubscribe` and `lift`, operators are to be implemented by extending the base classes. For example, the `map`
```

Comment:
That is why I showed that it can be done in a proof of concept project. That you reviewed.